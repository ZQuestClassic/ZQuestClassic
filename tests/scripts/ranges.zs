void test()
{
  int x = 5;

  switch (x) {
  case 0..5:
    printf("ERROR\n");
    break;
  case 5..10:
    printf("ERROR\n");
    break;
  default:
    printf("SUCCESS\n");
    break;
  }

  switch (x) {
  case 0=..5:
    printf("ERROR\n");
    break;
  case 5 =..10:
    printf("SUCCESS\n");
    break;
  default:
    printf("ERROR\n");
    break;
  }

  switch (x) {
  case 5..=10:
    printf("ERROR\n");
    break;
  case 0..=5:
    printf("SUCCESS\n");
    break;
  default:
    printf("ERROR\n");
    break;
  }

  switch (x) {
  case 0 =..=5:
    printf("SUCCESS\n");
    break;
  case 5 =..=10:
    printf("ERROR\n"); // because a higher case in this switch should've already
                       // matched
    break;
  default:
    printf("ERROR\n");
    break;
  }
}
