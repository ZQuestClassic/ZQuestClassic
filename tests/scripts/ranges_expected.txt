Compiling 'ranges.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
Pass 5: Generating object code
Pass 6: Assembling
Success!
Compile finished with exit code '0' (success)

QUIT